control-plane/csi-controller/src/controller.rs

```diff
@@ -532,16 +538,19 @@ impl rpc::csi::controller_server::Controller for CsiControllerSvc {
                 )));
             }
         } else {
-            // Volume is not published, bail out.
-            debug!(
-                "Volume {} is not published, not unpublishing",
-                args.volume_id
-            );
-            return Ok(Response::new(ControllerUnpublishVolumeResponse {}));
+            // Handle situation when the node that hosts nexus is offline and check
+            // volume status explicitly against 'Unknown'.
+            if volume.state.status != VolumeStatus::Unknown {
```

Comment:
we might be published here, we need to check what the spec says. We could be publish but the nexus is offline for some reason.